src/main/scala/com/gu/mobile/dynamo/backup/Lambda.scala

```diff
@@ -0,0 +1,83 @@
+package com.gu.mobile.dynamo.backup
+
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDate, LocalDateTime, ZoneId}
+import java.util.Date
+import java.util.concurrent.TimeUnit
+
+import com.amazonaws.auth.{AWSCredentialsProviderChain, DefaultAWSCredentialsProviderChain}
+import com.amazonaws.auth.profile.ProfileCredentialsProvider
+import com.amazonaws.regions.Regions.EU_WEST_1
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.amazonaws.services.dynamodbv2.model.{CreateBackupRequest, DeleteBackupRequest, ListBackupsRequest, ListBackupsResult}
+import com.gu.Logging
+
+import collection.JavaConverters._
+
+
+object Lambda extends Logging {
+
+  lazy val configuration: Configuration = {
+    logger.debug("Creating configuration")
+    new Configuration()
+  }
+
+  val credentials = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider("mobile"),
+    DefaultAWSCredentialsProviderChain.getInstance()
+  )
+
+  lazy val dynamoDBClient: AmazonDynamoDBClient = {
+    logger.debug("Creating dynamo db client")
+    new AmazonDynamoDBClient(credentials).withRegion(EU_WEST_1)
+  }
+
+  val daysToBackup = configuration.daysToBackup
+  val daysToBackupLong = daysToBackup - 1
+
+  val formatter = DateTimeFormatter.ofPattern("dd-MMM-yyyy")
+
+  def handler() : Unit = {
+
+     def maybeEmptyString2Option(s: String) = Option(s).filter(_.trim.isEmpty)
+
+    //Because aws uses java.util.date!
+     def toUtilDate(localDate: LocalDateTime) : Date = {
+      Date.from(localDate.atZone(ZoneId.systemDefault()).toInstant)
+    }
+
+    def makeListBackupsRequest(tableName: String, now: LocalDateTime) = new ListBackupsRequest()
+      .withTimeRangeLowerBound(toUtilDate(now.minusMinutes(120)))
+      .withTimeRangeUpperBound(toUtilDate(now.minusMinutes(15)))
+      .withTableName(tableName)
+
+
+    val now = LocalDateTime.now()
+
+    configuration.tables.map {
+       tableName =>
+         logger.info(s"Backing up $tableName")
+         val backupRequest = new CreateBackupRequest()
+           .withTableName(tableName)
+           .withBackupName(s"$tableName-backup-${now.format(formatter)}")
+         val backupResponse = dynamoDBClient.createBackup(backupRequest)
+         logger.info(s"Backuo completed succesfully for table $tableName")
+
+         val latestResponse = dynamoDBClient.listBackups(makeListBackupsRequest(tableName,now))
+
+         val latestBackupCount = latestResponse.getBackupSummaries.size
+         logger.info(s"Total backup count for table: $latestBackupCount")
+
+         if (latestBackupCount > daysToBackup) {
```

Comment:
Is there a way to have the logic deleting older backup based on their date rather than the total count of backup? Otherwise we could have a case where adding multiple backup the same day would delete older data.

Reply:
I'm assuming there'd be one back up a day, so it's kind of inference. Let's hava look and see if it can be improved though

Reply:
When you create the 'listBackUpSummariesRequest' object you supply a date range. Any backups will be from within that date-range.

Reply:
See lines 51-2